src/transform/common.jl

```diff
@@ -70,26 +69,24 @@ function maybe_rewrite_debuginfo!(ir::IRCode, settings::Settings)
 end
 
 function rewrite_debuginfo!(ir::IRCode)
-    debuginfo = ir.debuginfo
-    firstline = debuginfo.firstline
-    empty!(debuginfo.edges)
-    empty!(debuginfo.codelocs)
     for (i, stmt) in enumerate(ir.stmts)
-        push!(debuginfo.codelocs, i, i, 1)
-        inst = stmt[:inst]
         type = stmt[:type]
-        push!(debuginfo.edges, debuginfo_edge(i, inst, type))
+        annotation = type === nothing ? "" : " (inferred type: $type)"
+        # Work around `show` functions requiring `invokelatest` queries
+        # that may be problematic to execute from within generated functions.
+        local inst
+        try
+            inst = string(stmt[:inst])
+        catch e
+            isa(e, UndefVarError) && continue
+            rethrow()
+        end
```

Comment:
That is a bit of a hack, though calling `show` is kind of UB from generated functions. I'm only allowing `UndefVarError` here which are currently caused by binding lookups in certain cases, but it's not impossible that we later discover that other exceptions may be thrown.